src/os/vxworks/src/os-impl-common.c

```diff
@@ -192,8 +191,11 @@ int32 OS_VxWorks_GenericSemTake(SEM_ID vxid, int sys_ticks)
          * check for the timeout condition,
          * which has a different return code and
          * not necessarily an error of concern.
+         *
+         * vxworks7: if sys_ticks == 0, then if the semaphore can
+         * not be taken S_objLib_OBJ_UNAVAILABLE will be returned
          */
-        if (errno == S_objLib_OBJ_TIMEOUT)
+        if ((errno == S_objLib_OBJ_TIMEOUT) || (!sys_ticks && (errno == S_objLib_OBJ_UNAVAILABLE)))
```

Comment:
See my previous comment, can this be simplified to just:

`if (sys_ticks == 0 || errno == S_objLib_OBJ_TIMEOUT)`

Reply:
This doesn't work. errno can also be S_objLib_OBJ_ID_ERROR or S_intLib_NOT_ISR_CALLABLE.

Reply:
> This doesn't work. errno can also be S_objLib_OBJ_ID_ERROR or S_intLib_NOT_ISR_CALLABLE

But my point is - if sys_ticks is 0, does that detail matter?

Also - related question - Can `S_objLib_OBJ_UNAVAILABLE` get generated for other reasons too, when sys_ticks is nonzero?

Reply:
It can still be called with a bad object id or from an ISR context if sys_ticks==0. Should that ever happen? No. But we wouldn't want to return "everything is ok" if it does.